### PR TITLE
Address some warnings

### PR DESCRIPTION
**Pull request by mokagio**
I noticed a bunch of warnings while monitoring the warnings during my recent SwiftLint work at #20103 

## Regression Notes

1. Potential unintended areas of impact – N.A.
2. What I did to test those areas of impact (or what existing automated tests I relied on) – N.A.
3. What automated tests I added (or what prevented me from doing so) – N.A.

---

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes. **N.A.**
- [x] I have considered adding accessibility improvements for my changes. **N.A.**
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. **N.A.**

### CODE REVIEW

File: WordPress/WordPressTest/JetpackCapabilitiesServiceTests.swift [hunk 1]
```diff
@@ -23,8 +23,8 @@ class JetpackCapabilitiesServiceTests: CoreDataTestCase {
         let service = JetpackCapabilitiesService(coreDataStack: contextManager, capabilitiesServiceRemote: remoteMock)
 
         service.sync(blogs: [RemoteBlog.mock()], success: { blogs in
-            XCTAssertTrue(blogs.first!.capabilities["backup"] as! Bool)
-            XCTAssertTrue(blogs.first!.capabilities["scan"] as! Bool)
+            XCTAssertTrue(blogs.first!.capabilities["backup"]!)
+            XCTAssertTrue(blogs.first!.capabilities["scan"]!)
```
Comment (mokagio) [inline WordPress/WordPressTest/JetpackCapabilitiesServiceTests.swift:27]: `XCTAssertTrue` already expects a `Bool` as input, casting is (should be?) unnecessary. The compiler will (should?) figure it out.

File: WordPress/Classes/Stores/StatsRevampStore.swift [hunk 2]
```diff
@@ -383,12 +383,13 @@ private extension StatsRevampStore {
...
     func persistData<TimeIntervalType: StatsTimeIntervalData & TimeIntervalStatsRecordValueConvertible>(_ data: TimeIntervalType?) {
         guard
+            let data,
             let siteID = SiteStatsInformation.sharedInstance.siteID,
             let blog = Blog.lookup(withID: siteID, in: ContextManager.shared.mainContext) else {
             return
         }
 
-        data.flatMap { StatsRecord.record(from: $0, for: blog) }
+        _ = StatsRecord.record(from: data, for: blog)
```
Comment (staskus) [inline WordPress/Classes/Stores/StatsRevampStore.swift:392]: Thanks for fixing the warning. And yes, that is definitely a wrong application of `flatMap` 🤦‍♂️

File: WordPress/WordPressTest/JetpackCapabilitiesServiceTests.swift (hunk 1 above)
Comment (staskus) [reply WordPress/WordPressTest/JetpackCapabilitiesServiceTests.swift:27]: Correct, `public var capabilities: [String: Bool]` so no casting is needed.

**Review by staskus — APPROVED** [review]
Thanks for addressing the warnings 👍

**Pull request merged**